Sql history

davidhagan opened the pull request:
This brings an alternate history backend (sql driver with an apache db commons threadpool) to save checkresults, so that we can build a service health dashboard with historical data, and it also brings an update to lift, scala and some of the libraries.

dougecko commented on build.sbt:
```diff
@@ -30,29 +94,103 @@ javaOptions in container ++= Seq(
+    "com.sun.mail" % "javax.mail" % "1.6.2",
+    "com.hacklanta" %% "lift-formality_3.3" % "1.2.0",
+    "io.github.classgraph" % "classgraph" % "4.6.18",
+    "com.sksamuel.scrimage" % "scrimage-core_2.12" % "3.0.0-alpha4",
```
Do we need alpha dependencies?

dougecko commented on src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java:
```diff
@@ -0,0 +1,359 @@
+/*
```
Did you hack this from the original version?  (Why copy into the project?)

davidhagan commented:
yes.  Google's library is inadequate and unextendable, so I've had to hack
this one file.

On Fri, 7 Aug 2020, 07:41 Doug Hagan, <notifications@github.com> wrote:

> *@dougecko* commented on this pull request.
> ------------------------------
>
> In src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java
> <https://github.com/StackableRegiments/quacker/pull/1#discussion_r466701570>
> :
>
> > @@ -0,0 +1,359 @@
> +/*
>
> Did you hack this from the original version? (Why copy into the project?)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/StackableRegiments/quacker/pull/1#pullrequestreview-462895974>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYVB6LWGWZCRU4CGR2IZMDR7MPQXANCNFSM4PVQYD6A>
> .
>

davidhagan commented:
probably don't need this one.  will check.

On Fri, 7 Aug 2020, 07:38 Doug Hagan, <notifications@github.com> wrote:

> *@dougecko* commented on this pull request.
> ------------------------------
>
> In build.sbt
> <https://github.com/StackableRegiments/quacker/pull/1#discussion_r466700296>
> :
>
> > +    "org.specs2" %% "specs2-core" % "3.9.4" % "test",
> +    "com.h2database" % "h2" % "1.4.187",
> +    "com.github.dakatsuka" %% "akka-http-oauth2-client" % "0.1.0",
> +    "io.lemonlabs" %% "scala-uri" % "1.4.4",
> +    "com.google.api-client" % "google-api-client" % "1.25.0",
> +    "com.softwaremill.sttp" %% "core" % "1.5.0",
> +    "com.softwaremill.sttp" %% "okhttp-backend" % "1.5.11",
> +    "org.ekrich" %% "sconfig" % "0.8.0",
> +    "javax.servlet" % "javax.servlet-api" % "3.0.1" % "provided",
> +    "org.eclipse.jetty" % "jetty-webapp" % "9.4.11.v20180605",
> +    "net.rcarz" % "jira-client" % "0.6.3-IHTSDO",
> +    "javax.mail" % "javax.mail-api" % "1.6.2",
> +    "com.sun.mail" % "javax.mail" % "1.6.2",
> +    "com.hacklanta" %% "lift-formality_3.3" % "1.2.0",
> +    "io.github.classgraph" % "classgraph" % "4.6.18",
> +    "com.sksamuel.scrimage" % "scrimage-core_2.12" % "3.0.0-alpha4",
>
> Do we need alpha dependencies?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/StackableRegiments/quacker/pull/1#pullrequestreview-462894383>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYVB6MQ27VZSVPH7NET3Q3R7MPFBANCNFSM4PVQYD6A>
> .
>

davidhagan commented:
This branch just gained another feature - it is now possible to configure quacker to use github as the authentication domain (and also there's a mock authentication domain for local testing of IAM behaviours).  After all, what technical person who might need to see our component monitoring system wouldn't have a github account?  This means that there's still an unauthenticated behaviour, and you can still show some checks to unauthenticated people, but you can restrict other behaviours to users who've authenticated via github.

Also, I've added scripts for hosting it inside EKS as well, under the .kube directory, because I'm doing that on one of our deployments.  That doesn't change or break local usage, or local publishing, though I have changed the config directory to appConf, and I've added a local config file called appConf/application.xml to hold the authenticator's details - it never should be  hotswapped, after all.
